src/pages/home/report/LinkPreviewer.tsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import {Image, View} from 'react-native';
+import Text from '@components/Text';
+import TextLink from '@components/TextLink';
+import useStyleUtils from '@hooks/useStyleUtils';
+import useTheme from '@hooks/useTheme';
+import useThemeStyles from '@hooks/useThemeStyles';
+import variables from '@styles/variables';
+import type {LinkMetadata} from '@src/types/onyx/ReportAction';
+
+const IMAGE_TYPES = ['jpg', 'jpeg', 'png'];
+const MAX_IMAGE_HEIGHT = 180;
+const MAX_IMAGE_WIDTH = 340;
+
+type LinkPreviewerProps = {
+    /** Data about links provided in message. */
+    linkMetadata?: LinkMetadata[];
+
+    /** Maximum amount of visible link previews. -1 means unlimited amount of previews */
+    maxAmountOfPreviews?: number;
+};
+
+function LinkPreviewer({linkMetadata = [], maxAmountOfPreviews = -1}: LinkPreviewerProps) {
+    const theme = useTheme();
+    const styles = useThemeStyles();
+    const StyleUtils = useStyleUtils();
+    const uniqueLinks = linkMetadata.filter((link, index) => linkMetadata.findIndex(({url}) => url === link.url) === index);
```

Comment:
This is a slower implementation of uniqBy function.

Maybe use this

```js
const uniqBy = (arr: any[], predicate: (item: any) => string) => {
  const cb = typeof predicate === 'function' ? predicate : (o) => o[predicate];
  const result = [];
  const map = new Map();

  arr.forEach((item) => {
    const key = (item === null || item === undefined) ? item : cb(item);

    if (!map.has(key)) {
      map.set(key, item);
      result.push(item);
    }
  });

  return result;
};
```

Reply:
Indeed this is a better and faster implementation. But do we need something like this? linkMetadata is in 99.9% of times, a really small array(1 - 3 elements) and this optimization is not giving us that much(also _.uniqBy function is used only once in our codebase, so we won't be able to reuse it). That being said, can we keep the current implementation, as it seems to me that it's a little easier to understand?

Reply:
I agree that most of the cases won't have problem with the current implementation. But lets keep the optimised implementation as current implementation may introduce extra latencies, which we don't want

Reply:
changed filter to better implementation. How do you like it?

Reply:
@shubham1206agra seems like you forgot to press approve 😅

Reply:
Oh I just need to test this once. I will approve this today.